File: frontend/app/src/main/java/com/example/frontend_nhom1/model/Task.kt
package com.example.frontend_nhom1.model

data class Task(
    val _id: String,
    val title: String,
    val description: String? = null,
    val completed: Boolean = false,
    val owner: String? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null
)

File: backend/models/Task.js
const mongoose = require("mongoose");

const taskSchema = new mongoose.Schema(
    {
        title: { type: String, required: true, minlength: 1, maxlength: 200 },
        description: { type: String, maxlength: 2000 },
        completed: { type: Boolean, default: false },
        owner: {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true,
            index: true,
        },
    },
    { timestamps: true }
);

taskSchema.index({ owner: 1, createdAt: -1 });
taskSchema.index({ owner: 1, title: 1 });

module.exports = mongoose.model("Task", taskSchema);
